SCMOD-6865: Updated to recognise false positive pattern matches

Anthony-McGreevy opened the pull request:
https://portal.digitalsafe.net/browse/SCMOD-6865

Anthony-McGreevy committed:
Updated to recognise false positive pattern matches

dermot-hardy commented on worker-markup-core/src/main/java/com/github/cafdataprocessing/worker/markup/core/MarkupHeadersAndBody.java:
```diff
@@ -174,8 +174,9 @@ private void markUpHeadersAndBody(final Element parentElement)
-                    addCondensedHeader(nattyParser, headersElement, line, matcher);
+                    if(addCondensedHeader(nattyParser, headersElement, line, matcher)){
+                       bodyIndex++;
+                    }
```
Possible issue here as discussed.

dermot-hardy reviewed (changes requested):
Test case required for this.

Anthony-McGreevy committed:
Update

Anthony-McGreevy committed:
Merge branch 'develop' into SCMOD-6865

Anthony-McGreevy committed:
Updated to reflect comments

Anthony-McGreevy commented:
@dermot-hardy I have added the break after the check statement. I have also updated the new testcase to remove the expectation of a RuntimeException for this message.

dermot-hardy commented on worker-markup-core/src/test/java/com/github/cafdataprocessing/worker/markup/core/MarkupOfHeadersAndBodyTest.java:
```diff
@@ -110,11 +110,8 @@ public void testCondensedHeaderMultiLangMappings() throws IOException, JDOMExcep
-            assertTrue(ex.getMessage().equals("Logic error detected.  E-mail fidelity has been lost!"));
-        }
+        MarkupHeadersAndBody.markUpHeadersAndBody(xmlDocument, emailHeaderMappings, condensedHeaderMultiLangMappings);
+        //No assert nessesary, if this test had failed it would have thrown a RuntimeException.
```
The other tests show the output.  This one does not.

Anthony-McGreevy replied:
The other tests verify the marking up of certain sections of a header, this test is to verify no exception is thrown which is why it's not required.

Anthony-McGreevy committed:
Updated test

Anthony-McGreevy commented:
@dermot-hardy

dermot-hardy commented on worker-markup-core/src/test/resources/xml/FalsePositiveCondensedHeaderMultiLangMappingExpected.xml:
```diff
@@ -0,0 +1,35 @@
+-->
+<root>
+    <CONTENT>
+        <email>On a Friday night, December 15 2000, Congress passed complex and highly technical legislation which essentially ensured that much of the $94,000bn over-the-counter derivatives industry would operate outside the nation's commodities laws.
```
If this is the output then I'm glad I kept pressing for it because it would suggest that this is not working.  It is identical to the input.  Look at how the other expectations are different from the inputs.

Anthony-McGreevy replied:
added the element tags, they were being returned but not used in the verification. Added them to the expectation and now assert on an empty header block.

Anthony-McGreevy committed:
Updated expectation with xml element tags
